packages/analytics-js-integrations/src/integrations/GainsightPX/browser.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable lines-between-class-members */
+import {
+  NAME,
+  DISPLAY_NAME,
+} from '@rudderstack/analytics-js-common/constants/integrations/GainsightPX/constants';
+import Logger from '../../utils/logger';
+import { loadNativeSdk } from './nativeSdkLoader';
+
+const logger = new Logger(DISPLAY_NAME);
+
+class GainsightPX {
+  constructor(config, analytics, destinationInfo) {
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.analytics = analytics;
+    this.productKey = !config.productKey ? '' : config.productKey;
+    this.dataCenter = !config.dataCenter ? 'US' : config.dataCenter;
+    this.extraConfig = !config.extraConfig ? "{}" : config.extraConfig;
```

Comment:
Are these the extraConfig passed during SDK initialization?
https://support.gainsight.com/PX/Administration/General/Javascript_SDK_Configuration

```
{
    <!-- setting content security polic -->
    "fullDomainCookie": true
    "htmlSanitization": true,
    "htmlSanitizationAllowedDomains": ["https://allow.this.domain.com"]
    <!-- setting checksum endpoint -->
    "engagementChecksumFileUrl": "https://your.domain.com/checksums.json"
    }
  )
```

If yes, then you can configure these as separate settings in the integration config

Reply:
@Gauravudia - there are a lot (100) of possible values for the extra config settings.  We occasionally add new ones also.  For this reason, I don't think we should try to add them separately.

Reply:
If there are many extra config settings, Then we can use [integrations](https://www.rudderstack.com/docs/event-spec/standard-events/common-fields/) object for that.
example:-
https://github.com/rudderlabs/rudder-sdk-js/blob/9d4fd849e8f21675f1a84a14369fa57b5e83478f/packages/analytics-js-integrations/src/integrations/Vero/utils.js#L14

Reply:
Updated, removed extraConfig, changed to use integrationOptions